contracts/StakeManagerV2.sol

```diff
@@ -0,0 +1,343 @@
+        BNBX.transferFrom(msg.sender, address(this), _amount);
+        emit RequestedWithdrawal(msg.sender, _amount);
+
+        return requestId;
+    }
+
+    /// @notice Start the batch undelegation process.
+    /// @param _batchSize The size of the batch.
+    /// @param _operator The address of the operator to undelegate from.
+    function startBatchUndelegation(
+        uint256 _batchSize,
+        address _operator
+    )
+        external
+        override
+        whenNotPaused
+        nonReentrant
+        onlyRole(OPERATOR_ROLE)
+    {
+        if (_operator == address(0)) {
+            // if operator is not provided, use preferred operator
+            _operator = OPERATOR_REGISTRY.preferredWithdrawalOperator();
+        }
+        if (!OPERATOR_REGISTRY.operatorExists(_operator)) {
+            revert OperatorNotExisted();
+        }
+
+        address creditContract = STAKE_HUB.getValidatorCreditContract(_operator);
+        uint256 pooledBnb = IStakeCredit(creditContract).getPooledBNB(address(this));
+
+        uint256 cummulativeBnbXToBurn;
+        uint256 processedCount;
+        uint256 amountInBnbXToBurn;
+        while (firstUnprocessedUserIndex < withdrawalRequests.length && processedCount < _batchSize) {
+            cummulativeBnbXToBurn = amountInBnbXToBurn + withdrawalRequests[firstUnprocessedUserIndex].amountInBnbX;
+            if (pooledBnb >= convertBnbXToBnb(cummulativeBnbXToBurn)) {
+                amountInBnbXToBurn = cummulativeBnbXToBurn;
+                withdrawalRequests[firstUnprocessedUserIndex].processed = true;
+                withdrawalRequests[firstUnprocessedUserIndex].batchId = batchWithdrawalRequests.length;
+                processedCount++;
+                firstUnprocessedUserIndex++;
+            } else {
+                break;
+            }
+        }
+
+        if (amountInBnbXToBurn == 0) revert NoWithdrawalRequests();
+        uint256 amountToWithdrawFromOperator = convertBnbXToBnb(amountInBnbXToBurn);
+        totalDelegated -= amountToWithdrawFromOperator;
+
+        batchWithdrawalRequests.push(
+            BatchWithdrawalRequest({
+                amountInBnb: amountToWithdrawFromOperator,
+                amountInBnbX: amountInBnbXToBurn,
+                unlockTime: block.timestamp + STAKE_HUB.unbondPeriod(),
+                operator: _operator,
+                isClaimable: false
+            })
+        );
+
+        uint256 shares = IStakeCredit(creditContract).getSharesByPooledBNB(amountToWithdrawFromOperator);
+        BNBX.burn(address(this), amountInBnbXToBurn);
+        STAKE_HUB.undelegate(_operator, shares);
+
+        emit StartedBatchUndelegation(_operator, amountToWithdrawFromOperator, amountInBnbXToBurn);
+    }
+
+    /// @notice Complete the undelegation process.
+    /// @dev This function can only be called by an address with the OPERATOR_ROLE.
+    function completeBatchUndelegation() external override whenNotPaused nonReentrant onlyRole(OPERATOR_ROLE) {
```

Comment:
Don't need a privileged role here. Can be permissionless, right?

Reply:
yes, i think so. updated in #78